### DISCUSSION TITLE
TCP flow control and congestion control

### DISCUSSION THREAD

**Discussion by kimkh0884** [ANLAB-KAIST/KENSv3#80] [Project Q&A 2. Transport Layer (TCP)] [answered]
The google document of PA#3 implies that we have to implement flow control. According to what I learned from lecture, the receiver advertises the free buffer space using receive window field of TCP header. However, when I looked the test case using Wireshark, all the packets have window size of 51200.

The google document of PA#3 implies that we do not have to implement congestion control. However, when I looked the test case using Wireshark, I could see window size increases exponentially (1->2->4->8-> ... ) just like slow start phase of congestion control.

Can I get some explanations regarding this issues?

**Comment by Pusnow** [comment, marked as answer]
Yes, the counterpart ("adversary") binary uses congestion control, but you don't have to implement congestion control.
The test script will not check it.
